internal/server/manager.go

```diff
@@ -183,3 +257,10 @@ func addTypeInformationToObject(scheme *runtime.Scheme, obj client.Object) {
 		break
 	}
 }
+
+func isClusterSummary(summary *configv1alpha1.ClusterSummary) bool {
+	// TODO: I didn't get where to get this...
```

Comment:
I would call this "verifyLabelConfiguration". In general this should never happen unless user manually modifies a ClusterSummary.

Since, correctly, you want to check for this scenario I would also add a check on summary.Labels being nil.

Don't check on the ```projectsveltos.io/cluster-profile-name```.  Outside this method  you can get the ClusterProfile data calling 

configv1alpha1.GetProfileOwnerReference(summary)